src/libs/HttpUtils.js

```diff
@@ -98,7 +101,10 @@ function xhr(command, data, type = CONST.NETWORK.METHOD.POST, shouldUseSecure =
 
     let apiRoot = shouldUseSecure ? CONFIG.EXPENSIFY.SECURE_EXPENSIFY_URL : CONFIG.EXPENSIFY.URL_API_ROOT;
 
-    if (CONFIG.IS_IN_STAGING && shouldUseStagingServer) {
+    // If we are in native mobile apps, we dont have access to up-to-date Config so we need to only rely on the toggle switch
+    const nativeStagingSwitcher = (platform === CONST.OS.ANDROID || platform === CONST.OS.IOS) && shouldUseStagingServer;
+    const webStagingSwitcher = CONFIG.IS_IN_STAGING && shouldUseStagingServer;
```

Comment:
Maybe create a lib called `shouldUseStagingServer` with two files `index.native.js` and `index.js`. Then replace all of these with that lib.

If I am not correct, nativeStagingSwitcher is only used for native checks.

Reply:
Interesting, I think this abstraction would be a bit premature, I can be persuaded the other way too, but at the moment his precise logic is used only here.

Reply:
We discourage platform checks so I still think it is fine to create a small lib but I am fine with this. It might have one issue, let me check.

Reply:
@parasharrajat What issue did you have in mind?

Reply:
On, Mobile web `nativeStagingSwitcher` will be true and thus `webStagingSwitcher` will never be evaluated.

Reply:
Do you think this will be fine?

Reply:
> On, Mobile web nativeStagingSwitcher will be true and thus webStagingSwitcher will never be evaluated.

So lets have a look into this:
- if you are in staging web, you want to decide which API to use based on the toggle
    - That will be achieved in this case too since we will in the end decide based in the `shouldUseStagingServer` value
- in production the `nativeStagingSwitcher` will be false because the `shouldUseStagingServer` is always false there as no toggle is shown to change that
    - that means we would have to consider `webStagingSwitcher` in production otto and that will be false in mobile web since we can read the config so `CONFIG.IS_IN_STAGING` is false


I think we are fine then, do you agree @parasharrajat ?

Reply:
This logic is very confusing to me so I'd have to agree with moving it to separate modules to make this easier to read. 

Also I can't help but mention that fixing our staging build so that it actually uses the staging .env is the preferable fix. I'm not sure if there's anything blocking that but it might be worth considering dropping this change and just working towards that fix instead

Reply:
@arosiclair 
> Also I can't help but mention that fixing our staging build so that it actually uses the staging .env is the preferable fix. I'm not sure if there's anything blocking that but it might be worth considering dropping this change and just working towards that fix instead

This has been considered as a first thing but the way how we deploy is normal and standard procedure. We build just once, for staging using the production .env. This way when we run all the test on this build, fix all the issues, we then just put this build to production. We dont have to build again and risk some odd quirks when building the production and essentially giving our users untested app.

> This logic is very confusing to me so I'd have to agree with moving it to separate modules to make this easier to read.

I can add more comments to this, just not sure if adding this to a module of its own will help with clarity. Will additional comments be good enough in your opinion @arosiclair ?

Thank you for the review 🙇

Reply:
I was about to write the same thing. 

1. `nativeStagingSwitcher` should only be used for native all the time. 
2. `webStagingSwitcher` should only be used for web | Desktop all the time.  

A easier cleanup would be. 
1. Create a lib `shouldUseStagingServer`. 
```
 //index.native.js
function shouldUseStagingServer (shouldUseStagingServer){
 return shouldUseStagingServer;
}
```

```
 //index.js
function shouldUseStagingServer (shouldUseStagingServer){
 return CONFIG.IS_IN_STAGING && shouldUseStagingServer;
}
```

Reply:
@parasharrajat Thank you!

Updated with this solution, can you please give this a test?